operator/pkg/helmreconciler/reconciler.go

```diff
@@ -374,7 +374,8 @@ func (h *HelmReconciler) getCRHash(componentName string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return strings.Join([]string{crName, crNamespace, componentName}, "-"), nil
+
+	return strings.Join([]string{crName, crNamespace, componentName, h.restConfig.Host}, "-"), nil
```

Comment:
Open to suggestions other than `h.restConfig.Host` but the goal was to make the cache key unique per-cluster to allow concurrent use.